picker_wrapper.go

```diff
@@ -120,6 +123,9 @@ func (bp *pickerWrapper) pick(ctx context.Context, failfast bool, opts balancer.
 			bp.mu.Unlock()
 			select {
 			case <-ctx.Done():
+				if lastError != nil {
+					return nil, nil, lastError
```

Comment:
Context errors are converted to `Canceled`/`DeadlineExceeded`, but this will report `Unavailable`.
It's OK to add details to the error message, but we should keep the status code.

Reply:
Thanks for your suggestion. I will change it.

Reply:
@menghanl 
I change it. PTAL.